torch/distributed/tensor/parallel/style.py

```diff
@@ -344,5 +588,33 @@ class ColwiseParallel(ParallelStyle):
     We assume the input to be a replicated :class:`DTensor` and output to be a sharded :class:`torch.Tensor`.
     """
 
-    def __init__(self, _prepare_input=make_input_replicate_1d, _prepare_output=make_sharded_output_tensor) -> None:
-        super().__init__(_prepare_input, _prepare_output)
+    def __init__(
+        self,
+        _prepare_input=None,
+        _prepare_output=None,
+        *,
+        input_layouts=Replicate(),
+        output_layouts=Shard(-1),
+        use_local_output=True,
+    ) -> None:
```

Comment:
are you also adding the documentations to `ColwiseParallel` and `RowwiseParalell` about what type of modules we supported, and the `input_layouts`, `output_layouts`, `use_local_output` documentation?

Reply:
They are added here: https://github.com/pytorch/pytorch/pull/111177